tests/30-EdvRecipients.spec.js

```diff
@@ -0,0 +1,268 @@
+    inserted.jwe.protected.should.be.a('string');
+    inserted.jwe.recipients.should.be.an('array');
+    inserted.jwe.recipients.length.should.equal(1);
+    isRecipient({recipient: inserted.jwe.recipients[0]});
+    inserted.jwe.iv.should.be.a('string');
+    inserted.jwe.ciphertext.should.be.a('string');
+    inserted.jwe.tag.should.be.a('string');
+    inserted.content.should.deep.equal({someKey: 'someValue'});
+  });
+
+  it('should insert a document with 5 recipients', async () => {
+    const {keyAgreementKey} = mock.keys;
+    const client = await mock.createEdv();
+    const testId = await EdvClient.generateId();
+    const doc = {id: testId, content: {someKey: 'someValue'}};
+    // create the did keys then the recipients
+    const recipients = (await Promise.all([1, 2, 3, 4]
+      .map(() => mock.createKeyAgreementKey())))
+      .map(createRecipient);
+    // note: when passing recipients it is important to remember
+    // to pass in the document creator. EdvClient will use the
+    // EdvOwner by default as a recipient if there are no recipients
+    // being passed in, but will not if you explicitly pass in recipients.
+    recipients.unshift({header: {kid: keyAgreementKey.id, alg: JWE_ALG}});
+    const inserted = await client.insert(
+      {keyResolver, invocationSigner, doc, recipients});
+    should.exist(inserted);
+    inserted.should.be.an('object');
+    inserted.id.should.equal(testId);
+    inserted.sequence.should.equal(0);
+    inserted.indexed.should.be.an('array');
+    inserted.indexed.length.should.equal(1);
+    inserted.indexed[0].should.be.an('object');
+    inserted.indexed[0].sequence.should.equal(0);
+    inserted.indexed[0].hmac.should.be.an('object');
+    inserted.indexed[0].hmac.should.deep.equal({
+      id: client.hmac.id,
+      type: client.hmac.type
+    });
+    inserted.indexed[0].attributes.should.be.an('array');
+    inserted.jwe.should.be.an('object');
+    inserted.jwe.protected.should.be.a('string');
+    inserted.jwe.recipients.should.be.an('array');
+    // we should have 5 recipients including the EDV owner.
+    inserted.jwe.recipients.length.should.equal(5);
+    inserted.jwe.iv.should.be.a('string');
+    inserted.jwe.ciphertext.should.be.a('string');
+    inserted.jwe.tag.should.be.a('string');
+    inserted.content.should.deep.equal({someKey: 'someValue'});
+    inserted.jwe.recipients.forEach((recipient, index) => {
+      const expected = recipients[index].header;
+      // the curve should be a 25519 curve
+      expected.crv = 'X25519';
+      // key type should be an Octet Key Pair
+      expected.kty = 'OKP';
+      isRecipient({recipient, expected});
+    });
+  });
+
+  it('should enable a capability for a recipient', async function() {
+    const {keyAgreementKey} = mock.keys;
+    const client = await mock.createEdv();
+    const testId = await EdvClient.generateId();
+    const doc = {id: testId, content: {someKey: 'someValue'}};
+    const didKeys = await Promise.all(
```

Comment:
Can this be simplified?
```js
    const didKeys = [await mock.createKeyAgreementKey()];
```

Reply:
I like this alot more than the code in there currently so sure I made the change: https://github.com/digitalbazaar/edv-client/pull/26/commits/2f851b4f632828632ca44c6f5bcd8cf2712abd83